### PR TITLE
Support for npm install requirements

### PR DESCRIPTION
**Pull request by Sammers21**
* Design has been updated
* using qulice `0.18.19`(`new` allowed in cycles)
* NpmCommandsTest has been updated to ensure `npm install` works correctly

### CODE REVIEW

**Comment by Sammers21** [comment]
@yegor256 , can you review this PR please?

File: src/test/java/com/yegor256/npm/NpmCommandsTest.java [hunk 1]
```diff
@@ -33,38 +33,72 @@
...
+                url
             )
             .inheritIO()
             .start()
-            .waitFor();
-        Assume.assumeThat(code, Matchers.equalTo(0));
+            .waitFor(),
+            Matchers.equalTo(0)
+        );
+        Assume.assumeThat(
```
Comment (yegor256) [inline src/test/java/com/yegor256/npm/NpmCommandsTest.java]: @Sammers21 I think you need `Matchers.assertThat()` instead

**Comment by yegor256** [comment]
@Sammers21 I'm not sure your tests really work, check the comment above. You need `assertThat`, not `assumeThat`

File: src/test/java/com/yegor256/npm/NpmCommandsTest.java (hunk 1 above)
Comment (Sammers21) [reply src/test/java/com/yegor256/npm/NpmCommandsTest.java]: I guess you mean `MatcherAssert`

**Comment by Sammers21** [comment]
@yegor256 , done, can you have another look?

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@Sammers21 something is wrong with tests, see above

**Comment by Sammers21** [comment]
@yegor256, looks like rultor is using an old version of node.js and npm:
```
node.js version - v6.14.2
npm version - v3.10.10
```

On traivs we have:
```
node.js version - v10.16.0
npm version - v6.9.0
```

**Comment by yegor256** [comment]
@Sammers21 feel free to contribute to Rultor's Dockerfile: https://github.com/yegor256/rultor/blob/master/src/docker/Dockerfile

**Comment by yegor256** [comment]
@Sammers21 once the Dockerfile is updated there, a new Docker image will be available and we will able to run merge here again

**Comment by Sammers21** [comment]
@yegor256 , i guess now we can try to merge it again

**Comment by Sammers21** [comment]
@rultor merge

**Comment by Sammers21** [comment]
@yegor256, ping

**Comment by yegor256** [comment]
@rultor merge

**Comment by Sammers21** [comment]
@yegor256 , rultor is still using the old version:
```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
npm ERR! Linux 4.4.0-78-generic
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "publish" "--registry" "http://127.0.0.1:33927"
npm ERR! node v6.14.2
npm ERR! npm  v3.10.10
npm ERR! code ENEEDAUTH
npm ERR! need auth auth required for publishing
npm ERR! need auth You need to authorize this machine using `npm adduser`
npm ERR! Please include the following file with any support request:
npm ERR!     /home/r/repo/src/test/resources/simple-npm-project/npm-debug.log
Running com.yegor256.npm.NpmCommandsTest
[pool-1-thread-2] INFO com.yegor256.npm.NpmRegistry - Listening on port: 33927
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 1.227 sec <<< FAILURE! - in com.yegor256.npm.NpmCommandsTest
npmPublishAndInstallWorks(com.yegor256.npm.NpmCommandsTest)  Time elapsed: 1.227 sec  <<< FAILURE!
java.lang.AssertionError:
```

**Comment by yegor256** [comment]
@Sammers21 feel free to submit a new PR to Rultor Dockerfile

**Comment by yegor256** [comment]
@Sammers21 use this one: https://github.com/yegor256/rultor-image

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@Sammers21 sorry :(

**Comment by yegor256** [comment]
@Sammers21 the Docker image is not used here correctly. `.rultor.yml` specifies `yegor256/java8`. Let me try to fix it.

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor stop

**Comment by yegor256** [comment]
@rultor merge

**Comment by Sammers21** [comment]
@yegor256, there is an error with ruby gem installation, which is not related to the PR

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@Sammers21 thanks!

**Pull request merged**